src/dogroups.c

```diff
@@ -167,20 +174,24 @@ SEXP dogroups(SEXP dt, SEXP dtcols, SEXP groups, SEXP grpcols, SEXP jiscols, SEX
         case REALSXP :
           REAL(VECTOR_ELT(xSD,j))[0] = NA_REAL;
           break;
+        case CPLXSXP :
+          COMPLEX(VECTOR_ELT(xSD, j))[0].r = NA_REAL;
+          COMPLEX(VECTOR_ELT(xSD, j))[0].i = NA_REAL;
+          break;
         case STRSXP :
           SET_STRING_ELT(VECTOR_ELT(xSD,j),0,NA_STRING);
           break;
         case VECSXP :
           SET_VECTOR_ELT(VECTOR_ELT(xSD,j),0,R_NilValue);
           break;
```

Comment:
Should we nocov this branch (or set it to error even?)

IIUC hitting this branch would require first joining on `VECSXP` column in `on`, I don't think we have any plan to support list-to-list joins?

Reply:
don't no cov it.  I'd say don't error either because something somewhere might be using it.  I'd say just leave it as-is for now until it can be covered at a later date.